draft/index.html

```diff
@@ -517,12 +522,26 @@ <h3>Structure of a Reconciliation Query</h3>
 	    <dt><code>limit</code></dt>
             <dd>A limit on the number of candidates to return, which must be a positive integer;</dd>
 	    <dt><code>properties</code></dt>
-	    <dd>An array of objects, where each object maps a <a href='#properties'>property</a> identifier (in the <code>pid</code> field) 
-        to one or more <a>property values</a> (in the <code>v</code> field). These are used to further filter the set of candidates (similar to a WHERE clause in SQL),
-		by allowing clients to specify other attributes of entities that should match, beyond their name in the <code>query</code> field.
-		How reconciliation services handle this further restriction ("must match all properties" or "should match some") and how it affects the score, is up to the service.
-		A reconciliation service that supports properties SHOULD provide a <a>suggest service</a> for discovering these properties;</dd>
-          </dl>
+        <dd><p>An array of objects, where each object maps a <a href='#properties'>property</a> identifier (in the <code>pid</code> field) to one or more <a>property values</a> (in the <code>v</code> field).
+        These are used to further refine the list of candidates by allowing clients to specify other attributes of entities, beyond their name in the <code>query</code> field.
+        A reconciliation service that supports properties SHOULD provide a <a>suggest service</a> for discovering these properties.</p>
+        <p>In addition to <code>pid</code> and <code>v</code>, services MAY support the following optional fields that allow clients to specify the effect of each property on the resulting list of candidates.
+          If these fields are omitted, the exact behavior ("must match all", "should match some", etc.) is up to the service.</p>
+        <dl>
+          <dt><code>required</code></dt>
+          <dd>A boolean indicating if a match for the property is required for an entity to enter the list of candidates (i.e. acting like a filter or a WHERE clause in SQL)
+            or optional (i.e. only effecting the entity's rank in the list of candidates);</dd>
+          <dt><code>match_quantifier</code></dt>
+          <dd>A string to indicate which of the values in <code>v</code> to match. MUST be "any" (equivalent to boolean OR), "all" (equivalent to boolean AND), or "none" (equivalent to boolean NOT);</dd>
+          <dt><code>match_qualifier</code></dt>
```

Comment:
This feels like a solution in search of a problem. Are there use cases described which need this? I'm tempted to just play the YAGNI card here.

Reply:
So my attempt here with `match_qualifier`, in particular combined with the mention of EDTF (brought into the discussion by @thadguidry), was to address @paulgirard's use case from https://github.com/reconciliation-api/specs/issues/114 (which we also discussed in our [April meeting](https://etherpad.lobid.org/p/reconciliation-april-2023)). Paul, would this proposal here help with your use case?

Reply:
The `match_quantifier` justifies the possibility to add the same property id multiple time is the same request to create compound queries. It's an answer to #105  I think that the possibility to mention "any" or "all" is very important to tweak candidates selection.
The `match_qualifier` is definitely an very elegant answer to the date issue #114. I think this proposal is smart by letting recon services expose what kind of magic they propose and let any query decide to use it or not.